test/jdk/java/awt/EventDispatchThread/InterruptEDTTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2010, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+/*
+ * @test
+ * @key headful
+ * @bug 7184401
+ * @summary Verify that events are not lost when EDT is interrupted.
+ * @modules java.desktop/sun.awt
+ * @run main InterruptEDTTest
+ */
+
+public class InterruptEDTTest {
+
+    private static Frame frame = null;
+    private static volatile Thread edt = null;
+    private static volatile Robot robot = null;
+    private static volatile int xLocation;
+    private static volatile int yLocation;
+    private static volatile int width;
+    private static volatile int height;
+
+    public static void main(String args[]) throws Exception {
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(() -> {
+                edt = Thread.currentThread();
```

Comment:
Why this adding listener code etc in EDT?

Reply:
It is just the registering of frame to the listener is on EDT and this should not impact the invocation of the listener callback method at runtime.